### ISSUE TITLE
openshift_logging fails with ansible 2.3.0 (& 2.2.2.0)

### ISSUE THREAD

**Issue by etsauer** [openshift/openshift-ansible#3810]
#### Description

`byo/config.yml` fails with during `openshift_logging` role:

```
TASK [openshift_logging : Check for logging project already exists] **************************************************
ok: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : Create logging project] ********************************************************************
skipping: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : Create logging cert directory] *************************************************************
ok: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : include] ***********************************************************************************
included: /root/repository/openshift-ansible/roles/openshift_logging/tasks/generate_certs.yaml for master-0.l1.casl.rht-labs.com

TASK [openshift_logging : Checking for ca.key] ***********************************************************************
ok: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : Checking for ca.crt] ***********************************************************************
ok: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : Checking for ca.serial.txt] ****************************************************************
ok: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : Generate certificates] *********************************************************************
changed: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : Checking for signing.conf] *****************************************************************
ok: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : template] **********************************************************************************
changed: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : include] ***********************************************************************************
included: /root/repository/openshift-ansible/roles/openshift_logging/tasks/procure_server_certs.yaml for master-0.l1.casl.rht-labs.com
included: /root/repository/openshift-ansible/roles/openshift_logging/tasks/procure_server_certs.yaml for master-0.l1.casl.rht-labs.com
included: /root/repository/openshift-ansible/roles/openshift_logging/tasks/procure_server_certs.yaml for master-0.l1.casl.rht-labs.com

TASK [openshift_logging : Checking for kibana.crt] *******************************************************************
ok: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : Checking for kibana.key] *******************************************************************
ok: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : Trying to discover server cert variable name for kibana] ***********************************
skipping: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : Trying to discover the server key variable name for kibana] ********************************
skipping: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : Creating signed server cert and key for kibana] ********************************************
skipping: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : Copying server key for kibana to generated certs directory] ********************************
skipping: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : Copying Server cert for kibana to generated certs directory] *******************************
skipping: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : Checking for kibana-ops.crt] ***************************************************************
ok: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : Checking for kibana-ops.key] ***************************************************************
ok: [master-0.l1.casl.rht-labs.com]

TASK [openshift_logging : Trying to discover server cert variable name for kibana-ops] *******************************
fatal: [master-0.l1.casl.rht-labs.com]: FAILED! => {"failed": true, "msg": "The conditional check 'cert_info.hostnames is undefined and {{ cert_info.procure_component }}_crt is defined and {{ cert_info.procure_component }}_key is defined' failed. The error was: error while evaluating conditional (cert_info.hostnames is undefined and {{ cert_info.procure_component }}_crt is defined and {{ cert_info.procure_component }}_key is defined): 'kibana' is undefined\n\nThe error appears to have been in '/root/repository/openshift-ansible/roles/openshift_logging/tasks/procure_server_certs.yaml': line 12, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Trying to discover server cert variable name for {{ cert_info.procure_component }}\n  ^ here\nWe could be wrong, but this one looks like it might be an issue with\nmissing quotes.  Always quote template expression brackets when they\nstart a value. For instance:\n\n    with_items:\n      - {{ foo }}\n\nShould be written as:\n\n    with_items:\n      - \"{{ foo }}\"\n"}
        to retry, use: --limit @/root/repository/casl-ansible/playbooks/openshift/end-to-end.retry

PLAY RECAP ***********************************************************************************************************
app-node-0.l1.casl.rht-labs.com : ok=213  changed=19   unreachable=0    failed=0   
app-node-1.l1.casl.rht-labs.com : ok=214  changed=19   unreachable=0    failed=0   
dns-0.l1.casl.rht-labs.com : ok=46   changed=11   unreachable=0    failed=0   
infranode-0.l1.casl.rht-labs.com : ok=213  changed=19   unreachable=0    failed=0   
localhost                  : ok=50   changed=4    unreachable=0    failed=0   
master-0.l1.casl.rht-labs.com : ok=744  changed=97   unreachable=0    failed=1   
```

##### Version

```
# ansible --version
ansible 2.3.0
  config file = /root/.ansible.cfg
  configured module search path = Default w/o overrides

# git describe
openshift-ansible-3.6.10-1-17-gcff716d

```

##### Steps To Reproduce
1. add `openshift_hosted_logging_deploy=true` to inventory.
2. run `playbooks/byo/config.yml`

##### Additional Info

Ran using a `group_vars/OSEv3.yml` file:

```
---
deployment_type: openshift-enterprise
openshift_deployment_type: openshift-enterprise
openshift_master_default_subdomain: "apps.{{ env_id }}.{{ dns_domain }}"

openshift_cloudprovider_kind: openstack
... cloud provider....

openshift_hosted_logging_deploy: true
openshift_hosted_logging_storage_kind: dynamic

openshift_hosted_metrics_deploy: true
openshift_hosted_metrics_storage_kind: dynamic

openshift_node_kubelet_args:
  minimum-container-ttl-duration:
  - "10s"
  maximum-dead-containers-per-container:
  - "2"
  maximum-dead-containers:
  - "50"
  pods-per-core:
  - '10'
  max-pods:
  - '250'
  image-gc-high-threshold:
  - '80'
  image-gc-low-threshold:
  - '60'

```

cc: @sdodson @ewolinetz

**Comment by origliante** [comment]
Hi, should this work on openshift enterprise 3.5 with ansible version 3.5.60? I see:

TASK [openshift_logging : Trying to discover server cert variable name for kibana-ops] ************************************************
 [WARNING]: when statements should not include jinja2 templating delimiters such as {{ }} or {% %}. Found: cert_info.hostnames is
undefined and {{ cert_info.procure_component }}_crt is defined and {{ cert_info.procure_component }}_key is defined

fatal: [cw-vsdms-oem01.lseg.stockex.local]: FAILED! => {
    "failed": true
}

MSG:

The conditional check 'cert_info.hostnames is undefined and {{ cert_info.procure_component }}_crt is defined and {{ cert_info.procure_component }}_key is defined' failed. The error was: error while evaluating conditional (cert_info.hostnames is undefined and {{ cert_info.procure_component }}_crt is defined and {{ cert_info.procure_component }}_key is defined): Unable to look up a name or access an attribute in template string ({% if cert_info.hostnames is undefined and kibana-ops_crt is defined and kibana-ops_key is defined %} True {% else %} False {% endif %}).
Make sure your variable name does not contain invalid characters like '-': unsupported operand type(s) for -: 'StrictUndefined' and 'bool'

The error appears to have been in '/usr/share/ansible/openshift-ansible/roles/openshift_logging/tasks/procure_server_certs.yaml': line 12, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:


- name: Trying to discover server cert variable name for {{ cert_info.procure_component }}
  ^ here
We could be wrong, but this one looks like it might be an issue with
missing quotes.  Always quote template expression brackets when they
start a value. For instance:

    with_items:
      - {{ foo }}

Should be written as:

    with_items:
      - "{{ foo }}"


        to retry, use: --limit @/usr/share/ansible/openshift-ansible/playbooks/byo/openshift-cluster/openshift-logging.retry

PLAY RECAP ****************************************************************************************************************************
cw-vsdms-oei01.lseg.stockex.local : ok=1    changed=0    unreachable=0    failed=0
cw-vsdms-oei02.lseg.stockex.local : ok=1    changed=0    unreachable=0    failed=0
cw-vsdms-oem01.lseg.stockex.local : ok=35   changed=0    unreachable=0    failed=1
cw-vsdms-oen01.lseg.stockex.local : ok=1    changed=0    unreachable=0    failed=0
cw-vsdms-oen02.lseg.stockex.local : ok=1    changed=0    unreachable=0    failed=0
cw-vsdms-oen03.lseg.stockex.local : ok=1    changed=0    unreachable=0    failed=0
localhost                  : ok=7    changed=0    unreachable=0    failed=0

[root@cw-vsdms-oeb01 ~]#
[root@cw-vsdms-oeb01 ~]# ansible --version
ansible 2.3.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.5 (default, Aug  2 2016, 04:20:16) [GCC 4.8.5 20150623 (Red Hat 4.8.5-4)]
[root@cw-vsdms-oeb01 ~]#

**Comment by sdodson** [comment]
@origliante the fix for 1.5/3.5 is only in 3.5.63 and newer.

**Comment by origliante** [comment]
@sdodson many thanks! do you randomly know when >3.5.63 will it be released for enterprise?

**Comment by sdodson** [comment]
@origliante It should be May 16th, however I'll note if you're running into this you're not using ansible from the OCP repos and you should stick to using ansible from the OCP channels if possible. We carefully control which version of ansible goes in there so you shouldn't run into issues like this.

**Comment by origliante** [comment]
@sdodson yep I'm only on the redhat repos:

# rpm -qa | grep ansible
ansible-2.3.0.0-3.el7.noarch
openshift-ansible-filter-plugins-3.5.60-1.git.0.b6f77a6.el7.noarch
openshift-ansible-docs-3.5.60-1.git.0.b6f77a6.el7.noarch
openshift-ansible-lookup-plugins-3.5.60-1.git.0.b6f77a6.el7.noarch
openshift-ansible-callback-plugins-3.5.60-1.git.0.b6f77a6.el7.noarch
openshift-ansible-playbooks-3.5.60-1.git.0.b6f77a6.el7.noarch
openshift-ansible-3.5.60-1.git.0.b6f77a6.el7.noarch
openshift-ansible-roles-3.5.60-1.git.0.b6f77a6.el7.noarch

**Comment by sdodson** [comment]
2.3.0.0-3.el7 came from epel, you shouldn't need epel enabled in order to
install OCP.

On May 5, 2017 12:31 PM, "origliante" <notifications@github.com> wrote:

@sdodson <https://github.com/sdodson> yep I'm only on the redhat repos:
rpm -qa | grep ansible

ansible-2.3.0.0-3.el7.noarch
openshift-ansible-filter-plugins-3.5.60-1.git.0.b6f77a6.el7.noarch
openshift-ansible-docs-3.5.60-1.git.0.b6f77a6.el7.noarch
openshift-ansible-lookup-plugins-3.5.60-1.git.0.b6f77a6.el7.noarch
openshift-ansible-callback-plugins-3.5.60-1.git.0.b6f77a6.el7.noarch
openshift-ansible-playbooks-3.5.60-1.git.0.b6f77a6.el7.noarch
openshift-ansible-3.5.60-1.git.0.b6f77a6.el7.noarch
openshift-ansible-roles-3.5.60-1.git.0.b6f77a6.el7.noarch

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/openshift/openshift-ansible/issues/3810#issuecomment-299512383>,
or mute the thread

**Comment by jaryn** [comment]
I am facing similar problem. The ansible is not installed from epel.

```
ansible.noarch                           2.2.2.0-1.el7                            @RH7-RHAOS-3.5
openshift-ansible.noarch                        3.5.53-1.git.0.8ade9f2.el7        @RH7-RHAOS-3.5
openshift-ansible-callback-plugins.noarch       3.5.53-1.git.0.8ade9f2.el7        @RH7-RHAOS-3.5
openshift-ansible-docs.noarch                   3.5.53-1.git.0.8ade9f2.el7        @RH7-RHAOS-3.5
openshift-ansible-filter-plugins.noarch         3.5.53-1.git.0.8ade9f2.el7        @RH7-RHAOS-3.5
openshift-ansible-lookup-plugins.noarch         3.5.53-1.git.0.8ade9f2.el7        @RH7-RHAOS-3.5
openshift-ansible-playbooks.noarch              3.5.53-1.git.0.8ade9f2.el7        @RH7-RHAOS-3.5
openshift-ansible-roles.noarch                  3.5.53-1.git.0.8ade9f2.el7        @RH7-RHAOS-3.5
```

**Comment by sdodson** [comment]
@jaryn This should be fixed in openshift-ansible-3.5.63-1 and newer

**Comment by jaryn** [comment]
It is fixed in the puddle repo. Not yet in the signed repo.

**Comment by shah-zobair** [comment]
Got the same issue and succeeded with:

## yum downgrade ansible-2.2.1.0

**Comment by sdodson** [comment]
@shah-zobair This should no longer be a problem on master and release-1.5 branches, which version of openshift ansible do you have? either `rpm -q openshift-ansible` or `git describe` in your git repo.

**Comment by virtusademo** [comment]
Hi All, 

i am trying to the downgrade the ansible version , Current ansible version is 

[root@ip-172-31-25-0 ~]# ansible --version
ansible 2.2.3.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
[root@ip-172-31-25-0 ~]#

I want downgrade the ansible-2.2.0.0-1.el7.noarchor ansible 2.2.2.0 .

Whenever i am trying downgrade ansible , i got the following output.

]# yum downgrade ansible-2.2.0.0-1.el7.noarch
Loaded plugins: amazon-id, rhui-lb, search-disabled-repos
Resolving Dependencies
--> Running transaction check
---> Package ansible.noarch 0:2.2.0.0-1.el7 will be a downgrade
---> Package ansible.noarch 0:2.2.3.0-1.el7 will be erased
--> Finished Dependency Resolution
Error: Package: openshift-ansible-3.4.124-1.git.0.8bc631d.el7.noarch (@rhel-7-server-ose-3.4-rpms)
           Requires: ansible >= 2.2.2.0
           Removing: ansible-2.2.3.0-1.el7.noarch (@rhel-7-server-ose-3.4-rpms)
               ansible = 2.2.3.0-1.el7
           Downgraded By: ansible-2.2.0.0-1.el7.noarch (rhel-7-server-ose-3.4-rpms)
               ansible = 2.2.0.0-1.el7
           Available: ansible-2.2.1.0-2.el7.noarch (rhel-7-server-ose-3.4-rpms)
               ansible = 2.2.1.0-2.el7
           Available: ansible-2.3.1.0-3.el7.noarch (rhel-7-server-extras-rpms)
               ansible = 2.3.1.0-3.el7
           Available: ansible-2.3.2.0-2.el7.noarch (rhel-7-server-extras-rpms)
               ansible = 2.3.2.0-2.el7
**********************************************************************
yum can be configured to try to resolve such errors by temporarily enabling
disabled repos and searching for missing dependencies.
To enable this functionality please set 'notify_only=0' in /etc/yum/pluginconf.d/search-disabled-repos.conf
**********************************************************************

Error: Package: openshift-ansible-3.4.124-1.git.0.8bc631d.el7.noarch (@rhel-7-server-ose-3.4-rpms)
           Requires: ansible >= 2.2.2.0
           Removing: ansible-2.2.3.0-1.el7.noarch (@rhel-7-server-ose-3.4-rpms)
               ansible = 2.2.3.0-1.el7
           Downgraded By: ansible-2.2.0.0-1.el7.noarch (rhel-7-server-ose-3.4-rpms)
               ansible = 2.2.0.0-1.el7
           Available: ansible-2.2.1.0-2.el7.noarch (rhel-7-server-ose-3.4-rpms)
               ansible = 2.2.1.0-2.el7
           Available: ansible-2.3.1.0-3.el7.noarch (rhel-7-server-extras-rpms)
               ansible = 2.3.1.0-3.el7
           Available: ansible-2.3.2.0-2.el7.noarch (rhel-7-server-extras-rpms)
               ansible = 2.3.2.0-2.el7
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

++++++++++++++++++++++++++++++++++++++++++++++

#yum downgrade ansible-2.3.1.0-3.el7.noarch
#yum downgrade ansible-2.2.3.0-1.el7.noarch 

the above downgrade completed .Problem is coming  yum downgrade ansible-2.2.0.0-1.el7.noarch .it is not downgradeing. 

Please help on these .

I am creating setup on opesnhift container platform 3.4 , Os version rhel 7.3 in aws servers.

[root@ip-172-31-25-0 ~]# ansible --version
ansible 2.2.3.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
[root@ip-172-31-25-0 ~]# rpm -qa | grep ansible
openshift-ansible-callback-plugins-3.4.124-1.git.0.8bc631d.el7.noarch
openshift-ansible-playbooks-3.4.124-1.git.0.8bc631d.el7.noarch
ansible-2.2.3.0-1.el7.noarch
openshift-ansible-3.4.124-1.git.0.8bc631d.el7.noarch
openshift-ansible-lookup-plugins-3.4.124-1.git.0.8bc631d.el7.noarch
openshift-ansible-roles-3.4.124-1.git.0.8bc631d.el7.noarch
openshift-ansible-docs-3.4.124-1.git.0.8bc631d.el7.noarch
openshift-ansible-filter-plugins-3.4.124-1.git.0.8bc631d.el7.noarch
[root@ip-172-31-25-0 ~]#

**Comment by virtusademo** [comment]
Hi All, 

i am trying to the downgrade the ansible version , Current ansible version is 

[root@ip-172-31-25-0 ~]# ansible --version
ansible 2.2.3.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
[root@ip-172-31-25-0 ~]#

I want downgrade the ansible-2.2.0.0-1.el7.noarchor ansible 2.2.2.0 .

Whenever i am trying downgrade ansible , i got the following output.

]# yum downgrade ansible-2.2.0.0-1.el7.noarch
Loaded plugins: amazon-id, rhui-lb, search-disabled-repos
Resolving Dependencies
--> Running transaction check
---> Package ansible.noarch 0:2.2.0.0-1.el7 will be a downgrade
---> Package ansible.noarch 0:2.2.3.0-1.el7 will be erased
--> Finished Dependency Resolution
Error: Package: openshift-ansible-3.4.124-1.git.0.8bc631d.el7.noarch (@rhel-7-server-ose-3.4-rpms)
           Requires: ansible >= 2.2.2.0
           Removing: ansible-2.2.3.0-1.el7.noarch (@rhel-7-server-ose-3.4-rpms)
               ansible = 2.2.3.0-1.el7
           Downgraded By: ansible-2.2.0.0-1.el7.noarch (rhel-7-server-ose-3.4-rpms)
               ansible = 2.2.0.0-1.el7
           Available: ansible-2.2.1.0-2.el7.noarch (rhel-7-server-ose-3.4-rpms)
               ansible = 2.2.1.0-2.el7
           Available: ansible-2.3.1.0-3.el7.noarch (rhel-7-server-extras-rpms)
               ansible = 2.3.1.0-3.el7
           Available: ansible-2.3.2.0-2.el7.noarch (rhel-7-server-extras-rpms)
               ansible = 2.3.2.0-2.el7
**********************************************************************
yum can be configured to try to resolve such errors by temporarily enabling
disabled repos and searching for missing dependencies.
To enable this functionality please set 'notify_only=0' in /etc/yum/pluginconf.d/search-disabled-repos.conf
**********************************************************************

Error: Package: openshift-ansible-3.4.124-1.git.0.8bc631d.el7.noarch (@rhel-7-server-ose-3.4-rpms)
           Requires: ansible >= 2.2.2.0
           Removing: ansible-2.2.3.0-1.el7.noarch (@rhel-7-server-ose-3.4-rpms)
               ansible = 2.2.3.0-1.el7
           Downgraded By: ansible-2.2.0.0-1.el7.noarch (rhel-7-server-ose-3.4-rpms)
               ansible = 2.2.0.0-1.el7
           Available: ansible-2.2.1.0-2.el7.noarch (rhel-7-server-ose-3.4-rpms)
               ansible = 2.2.1.0-2.el7
           Available: ansible-2.3.1.0-3.el7.noarch (rhel-7-server-extras-rpms)
               ansible = 2.3.1.0-3.el7
           Available: ansible-2.3.2.0-2.el7.noarch (rhel-7-server-extras-rpms)
               ansible = 2.3.2.0-2.el7
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

++++++++++++++++++++++++++++++++++++++++++++++

#yum downgrade ansible-2.3.1.0-3.el7.noarch
#yum downgrade ansible-2.2.3.0-1.el7.noarch 

the above downgrade completed .Problem is coming  yum downgrade ansible-2.2.0.0-1.el7.noarch .it is not downgradeing. 

Please help on these .

I am creating setup on opesnhift container platform 3.4 , Os version rhel 7.3 in aws servers.

[root@ip-172-31-25-0 ~]# ansible --version
ansible 2.2.3.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
[root@ip-172-31-25-0 ~]# rpm -qa | grep ansible
openshift-ansible-callback-plugins-3.4.124-1.git.0.8bc631d.el7.noarch
openshift-ansible-playbooks-3.4.124-1.git.0.8bc631d.el7.noarch
ansible-2.2.3.0-1.el7.noarch
openshift-ansible-3.4.124-1.git.0.8bc631d.el7.noarch
openshift-ansible-lookup-plugins-3.4.124-1.git.0.8bc631d.el7.noarch
openshift-ansible-roles-3.4.124-1.git.0.8bc631d.el7.noarch
openshift-ansible-docs-3.4.124-1.git.0.8bc631d.el7.noarch
openshift-ansible-filter-plugins-3.4.124-1.git.0.8bc631d.el7.noarch
[root@ip-172-31-25-0 ~]#

**Comment by sdodson** [comment]
@virtusademo you shouldn't have to downgrade the current branch should work fine with ansible 2.2.3.0 or 2.3.x too.
